src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,45 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private CopyOnWriteArrayList<EventListener> events = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor;
 
+    public EventManager() {
+        this.events = new CopyOnWriteArrayList<>();
+        this.executor = Executors.newCachedThreadPool();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        events.add(listener);
+    }
 
+    public void deregisterListener(EventListener listener) {
+        events.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : events) {
+            executor.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executor.shutdown();
+        try {
+            if (!executor.awaitTermination(60, TimeUnit.SECONDS)) {
```

Comment:
When shutting down the executor, the awaitTermination is called with a timeout of 60 seconds. It's important to ensure that this timeout is sufficient for all tasks to complete to avoid abrupt termination of processing. If tasks might take longer, consider adjusting the timeout value.